modules/apps/transfer/types/forwarding.go

```diff
@@ -8,26 +8,52 @@ import (
 
 const MaximumNumberOfForwardingHops = 16 // denotes the maximum number of forwarding hops allowed
 
-// NewForwarding creates a new Forwarding instance given a memo and a variable number of hops.
-func NewForwarding(memo string, hops ...Hop) Forwarding {
+// NewForwarding creates a new Forwarding instance given an unwind value and a variable number of hops.
+func NewForwarding(unwind bool, hops ...Hop) Forwarding {
 	return Forwarding{
-		Memo: memo,
-		Hops: hops,
+		Unwind: unwind,
+		Hops:   hops,
 	}
 }
 
 // Validate performs a basic validation of the Forwarding fields.
 func (fi Forwarding) Validate() error {
 	if len(fi.Hops) > MaximumNumberOfForwardingHops {
```

Comment:
Nit (can be done in a follow-up, no need to block this PR): Since this `hops` validation is also done in `ForwardingPacketData` `Validate`, maybe it could be extracted into a separate function, but still care of slight differences in the error messages (e.g. `forwarding` vs `forwarding packet data`), if we want that.